src/http.c

```diff
@@ -394,16 +394,14 @@ static mime_map_t map[] = { {".html", "text/html"},
 
 const char *filename_to_mimetype(const char *filename)
 {
-        int i = 0;
-        while (map[i].ext) {
+        for (int i = 0; map[i].ext != NULL; i++) {
                 int n = strlen(map[i].ext);
                 int m = strlen(filename);
                 if (m < n + 1)
                         continue;
```

Comment:
The continue statement here by-passes the i++ statement at the end of the block, and the code is stuck in a infinite loop.